config/experimental/config/doc.go

```diff
@@ -0,0 +1,19 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package config under config/experimental contains configuration related types and interfaces
+// that typically live under the "go.opentelemetry.io/collector/config" package but aren't stable yet
+// to be published there.
+// ATTENTION: the package is still experimental and subject to changes without advanced notice.
+package config
```

Comment:
Maybe we can call this configsource for now?

Reply:
I don't have a strong opinion about this just trying to see a path with fewer renames when/if this gets out of experimental.

Reply:
The initial PR had everything under the `configsource` package - but because of some renames to avoid stutter, it was suggested to move some parts to packages akin to their final destinations.